src/main/scala/edu/ie3/mobsim/io/probabilities/TripProbabilities.scala

```diff
@@ -32,3 +47,113 @@ final case class TripProbabilities(
     poiTransition: PoiTransition,
     tripDistance: TripDistance
 )
+
+object TripProbabilities extends LazyLogging {
+
+  def read(
+      pathsAndSources: PathsAndSources,
+      colSep: String
+  ): TripProbabilities = {
+
+    val firstDepartureOfDay = FirstDepartureFactory.getFromFile(
+      pathsAndSources.firstDepartureOfDayPath,
+      colSep
+    ) match {
+      case Failure(exception) =>
+        throw SourceException(
+          "Unable to get probabilities for first departure of day from path.",
+          exception
+        )
+      case Success(value) => value
+    }
+
```

Comment:
There's many copies of almost exactly this code in here, maybe we can find an abstraction for this e.g. in form of a common method

Reply:
One suggestion would be to actually throw the exceptions from within the `getFromFile` methods, so that we can catch them here and re-throw them within a `SourceException`. Pretty old school, but effective. Could use checked exceptions if one is concerned about safety here.

Reply:
#97